server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -3545,7 +3546,16 @@ public void startRecovery(
                 executeRecovery("from store", recoveryState, recoveryListener, this::recoverFromStore);
                 break;
             case REMOTE_STORE:
-                executeRecovery("from remote store", recoveryState, recoveryListener, l -> restoreFromRemoteStore(l));
+                // When remote indices are restored via remote metadata,
+                // the recovery source is REMOTE_STORE and unassigned Reason is CLUSTER_RECOVERED
+                // During remote index restore from local disk metadata, the unassigned Reason is still CLUSTER_RECOVERED
+                // but the RecoverySource is not REMOTE_STORE but EXISTING_STORE
```

Comment:
can the RecoverySource be EXISTING_STORE instead of REMOTE_STORE. change it to REMOTE_STORE when restore is triggered via API. Then, this special handling will not be needed.